// symbol-qr-library/dist/index.js
"use strict";
/**
 * (C) Symbol Contributors 2022
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QRCodeGenerator = exports.CosignatureSignedTransactionQR = exports.SignedTransactionQR = exports.MnemonicQR = exports.CosignatureQR = exports.TransactionQR = exports.ObjectQR = exports.AddressQR = exports.ContactQR = exports.AccountQR = exports.EncryptionService = exports.EncryptedPayload = exports.CosignatureSignedTransactionDataSchema = exports.SignedTransactionDataSchema = exports.RequestCosignatureDataSchema = exports.RequestTransactionDataSchema = exports.ExportObjectDataSchema = exports.ExportMnemonicDataSchema = exports.ExportAccountDataSchema = exports.AddContactDataSchema = exports.QRCodeDataSchema = exports.QRCode = exports.QRCodeStreamType = exports.QRCodeSettings = exports.QRCodeType = void 0;
__exportStar(require("./src/sdk"), exports);
var QRCodeType_1 = require("./src/QRCodeType");
Object.defineProperty(exports, "QRCodeType", { enumerable: true, get: function () { return QRCodeType_1.QRCodeType; } });
var QRCodeSettings_1 = require("./src/QRCodeSettings");
Object.defineProperty(exports, "QRCodeSettings", { enumerable: true, get: function () { return QRCodeSettings_1.QRCodeSettings; } });
var QRCodeStreamType_1 = require("./src/QRCodeStreamType");
Object.defineProperty(exports, "QRCodeStreamType", { enumerable: true, get: function () { return QRCodeStreamType_1.QRCodeStreamType; } });
// abstract QRCode
var QRCode_1 = require("./src/QRCode");
Object.defineProperty(exports, "QRCode", { enumerable: true, get: function () { return QRCode_1.QRCode; } });
// QR Code data schemas
var QRCodeDataSchema_1 = require("./src/QRCodeDataSchema");
Object.defineProperty(exports, "QRCodeDataSchema", { enumerable: true, get: function () { return QRCodeDataSchema_1.QRCodeDataSchema; } });
var AddContactDataSchema_1 = require("./src/schemas/AddContactDataSchema");
Object.defineProperty(exports, "AddContactDataSchema", { enumerable: true, get: function () { return AddContactDataSchema_1.AddContactDataSchema; } });
var ExportAccountDataSchema_1 = require("./src/schemas/ExportAccountDataSchema");
Object.defineProperty(exports, "ExportAccountDataSchema", { enumerable: true, get: function () { return ExportAccountDataSchema_1.ExportAccountDataSchema; } });
var ExportMnemonicDataSchema_1 = require("./src/schemas/ExportMnemonicDataSchema");
Object.defineProperty(exports, "ExportMnemonicDataSchema", { enumerable: true, get: function () { return ExportMnemonicDataSchema_1.ExportMnemonicDataSchema; } });
var ExportObjectDataSchema_1 = require("./src/schemas/ExportObjectDataSchema");
Object.defineProperty(exports, "ExportObjectDataSchema", { enumerable: true, get: function () { return ExportObjectDataSchema_1.ExportObjectDataSchema; } });
var RequestTransactionDataSchema_1 = require("./src/schemas/RequestTransactionDataSchema");
Object.defineProperty(exports, "RequestTransactionDataSchema", { enumerable: true, get: function () { return RequestTransactionDataSchema_1.RequestTransactionDataSchema; } });
var RequestCosignatureDataSchema_1 = require("./src/schemas/RequestCosignatureDataSchema");
Object.defineProperty(exports, "RequestCosignatureDataSchema", { enumerable: true, get: function () { return RequestCosignatureDataSchema_1.RequestCosignatureDataSchema; } });
var SignedTransactionDataSchema_1 = require("./src/schemas/SignedTransactionDataSchema");
Object.defineProperty(exports, "SignedTransactionDataSchema", { enumerable: true, get: function () { return SignedTransactionDataSchema_1.SignedTransactionDataSchema; } });
var CosignatureSignedTransactionDataSchema_1 = require("./src/schemas/CosignatureSignedTransactionDataSchema");
Object.defineProperty(exports, "CosignatureSignedTransactionDataSchema", { enumerable: true, get: function () { return CosignatureSignedTransactionDataSchema_1.CosignatureSignedTransactionDataSchema; } });
// encryption
var EncryptedPayload_1 = require("./src/EncryptedPayload");
Object.defineProperty(exports, "EncryptedPayload", { enumerable: true, get: function () { return EncryptedPayload_1.EncryptedPayload; } });
var EncryptionService_1 = require("./src/services/EncryptionService");
Object.defineProperty(exports, "EncryptionService", { enumerable: true, get: function () { return EncryptionService_1.EncryptionService; } });
// specialized QR Code classes
var AccountQR_1 = require("./src/AccountQR");
Object.defineProperty(exports, "AccountQR", { enumerable: true, get: function () { return AccountQR_1.AccountQR; } });
var ContactQR_1 = require("./src/ContactQR");
Object.defineProperty(exports, "ContactQR", { enumerable: true, get: function () { return ContactQR_1.ContactQR; } });
var AddressQR_1 = require("./src/AddressQR");
Object.defineProperty(exports, "AddressQR", { enumerable: true, get: function () { return AddressQR_1.AddressQR; } });
var ObjectQR_1 = require("./src/ObjectQR");
Object.defineProperty(exports, "ObjectQR", { enumerable: true, get: function () { return ObjectQR_1.ObjectQR; } });
var TransactionQR_1 = require("./src/TransactionQR");
Object.defineProperty(exports, "TransactionQR", { enumerable: true, get: function () { return TransactionQR_1.TransactionQR; } });
var CosignatureQR_1 = require("./src/CosignatureQR");
Object.defineProperty(exports, "CosignatureQR", { enumerable: true, get: function () { return CosignatureQR_1.CosignatureQR; } });
var MnemonicQR_1 = require("./src/MnemonicQR");
Object.defineProperty(exports, "MnemonicQR", { enumerable: true, get: function () { return MnemonicQR_1.MnemonicQR; } });
var SignedTransactionQR_1 = require("./src/SignedTransactionQR");
Object.defineProperty(exports, "SignedTransactionQR", { enumerable: true, get: function () { return SignedTransactionQR_1.SignedTransactionQR; } });
var CosignatureSignedTransactionQR_1 = require("./src/CosignatureSignedTransactionQR");
Object.defineProperty(exports, "CosignatureSignedTransactionQR", { enumerable: true, get: function () { return CosignatureSignedTransactionQR_1.CosignatureSignedTransactionQR; } });
// factory
var QRCodeGenerator_1 = require("./src/QRCodeGenerator");
Object.defineProperty(exports, "QRCodeGenerator", { enumerable: true, get: function () { return QRCodeGenerator_1.QRCodeGenerator; } });
